### PR TITLE
Update azure poetry lock

### PR DESCRIPTION
**Pull request by dmytro-afanasiev**
Add dependencies for:
- [azure.stream-job](https://github.com/cloud-custodian/cloud-custodian/pull/9118)

Update dependencies for:
- [azure.eventhub.filters.private-endpoint-connections](https://github.com/cloud-custodian/cloud-custodian/pull/9037)

### CODE REVIEW

File: tools/c7n_azure/pyproject.toml [hunk 1]
```diff
@@ -40,7 +40,7 @@ azure-mgmt-datafactory = "^1.0.0"
 azure-mgmt-desktopvirtualization = "^1.0.0"
 azure-mgmt-dns = "8.0.0b1"
 azure-mgmt-eventgrid = "^8.0.0"
-azure-mgmt-eventhub = "^8.0.0"
+azure-mgmt-eventhub = "11.0.0"
```
Comment (kapilt) [inline tools/c7n_azure/pyproject.toml:43]: version markers shouldn't be hard pins ideally, they should be carrot ideally re ^

Comment (kapilt) [reply tools/c7n_azure/pyproject.toml:43]: so that we can get updates, for releases we freeze the dependency graph in the wheels.

Comment (dmytro-afanasiev) [reply tools/c7n_azure/pyproject.toml]: Ok, i added a caret symbol

**Review by kapilt — APPROVED** [review]
lgtm, thanks

**Pull request merged**